geps/gep-2895/index.md

```diff
@@ -0,0 +1,250 @@
+    // (See https://www.rfc-editor.org/rfc/rfc7230#section-2.7.3).
+    //
+    // Input:
+    //   GET /foo?my-parameter=foo HTTP/1.1
+    //
+    // Config:
+    //   set:
+    //   - name: "my-parameter"
+    //     value: "bar"
+    //
+    // Output:
+    //   GET /foo?my-parameter=bar HTTP/1.1
+    //
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=name
+    // +kubebuilder:validation:MaxItems=16
+    Set []HTTPHeader `json:"set,omitempty"`
+
+    // Add adds the given query param(s) (name, value) to the HTTP request
+    // before the action.
+    //
+    // Input:
+    //   GET /foo?my-parameter=foo HTTP/1.1
+    //
+    // Config:
+    //   add:
+    //   - name: "my-parameter"
+    //     value: "bar"
+    //
+    // Output:
+    //   GET /foo?my-parameter=foo&my-parameter=bar HTTP/1.1
+    //
+    // +optional
+    // +listType=map
+    // +listMapKey=name
+    // +kubebuilder:validation:MaxItems=16
+    Add []HTTPHeader `json:"add,omitempty"`
+
+    // Remove the given query param(s) from the HTTP request before the action.
+    // The value of Remove is a list of query param names. Note that the query
+    // param names are case-sensitive (See
+    // https://www.rfc-editor.org/rfc/rfc7230#section-2.7.3).
+    //
+    // Input:
+    //   GET /foo?my-parameter1=foo&my-parameter2=bar&my-parameter3=baz HTTP/1.1
+    //
+    // Config:
+    //   remove: ["my-parameter1", "my-parameter3"]
+    //
+    // Output:
+    //   GET /foo?my-parameter2=bar HTTP/1.1
+    //
+    // +optional
+    // +listType=set
+    // +kubebuilder:validation:MaxItems=16
+    Remove []string `json:"remove,omitempty"`
+}
+```
+
+## Examples
+
+The following example shows how a HTTPRoute modifies the query parameter of an HTTP request before it is sent to the upstream target.
+
+It allows to add query parameter for only a certain canary backend, which can help in identifying certain users by the backend service. 
+Based on the following http rule, query parameter "passtoken=$sign_passtoken_plain" will be added to the requests be matched against the query parameter "gray=3", then the request will be routed to the canary service "http-route-canary:80".
+
+```yaml
+apiVersion: gateway.networking.k8s.io/v1
+kind: HTTPRoute
+metadata:
+  name: http-route-query
+spec:
+  hostnames:
+  - http.route.query.com
+  - http.route.queries.com
+  parentRefs:
+  - group: gateway.networking.k8s.io
+    kind: Gateway
+    name: http-gateway
+  rules:
+  - backendRefs:
+    - kind: Service
+      name: http-route-production
+      port: 80
+    matches:
+    - path:
+        type: PathPrefix
+        value: /
+  - backendRefs:
+    - kind: Service
+      name: http-route-canary
+      port: 80
+    filters:
+    - queryParamModifier:
+        add:
+        - name: passtoken
+          value: $sign_passtoken_plain
+      type: QueryParamModifier
+    matches:
+    - queryParams:
+      - name: gray
+        type: Exact
+        value: 3
+
+```
+
+## Prior Art
```

Comment:
```suggestion
## Implementation-specific solutions
```
I wouldn't use the "Prior art" term, as it is not a community-based prior art, but instead an implementation-specific effort.